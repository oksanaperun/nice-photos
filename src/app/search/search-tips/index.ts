export * from './search-tips.component';
