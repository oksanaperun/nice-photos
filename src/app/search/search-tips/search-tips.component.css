.search-tips {
  font-family: 'Nunito', serif;
  margin: 0 auto;
  width: 50%;
}

h4 {
  text-align: center;
  color: #673ab7;
  margin-bottom: 5px;
}

ul {
  font-size: small;
  margin: 0 auto;
  width: 65%;
}

li {
  color: #8e58f2;
}
