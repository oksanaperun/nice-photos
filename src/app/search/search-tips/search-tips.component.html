<div class="search-tips">
  <h4 i18n>Search tips:</h4>
  <ul>
    <li *ngFor="let tip of searchTips">{{tip}}</li>
  </ul>
</div>
