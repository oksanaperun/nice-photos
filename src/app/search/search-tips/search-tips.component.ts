import { Component, ChangeDetectionStrategy } from '@angular/core';

@Component({
  selector: 'app-search-tips',
  templateUrl: './search-tips.component.html',
  styleUrls: ['./search-tips.component.css'],
  changeDetection: ChangeDetectionStrategy.OnPush,
})

export class SearchTipsComponent { }
