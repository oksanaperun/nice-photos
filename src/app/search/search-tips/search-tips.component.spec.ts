import { TestBed, ComponentFixture } from '@angular/core/testing';
import { SearchTipsComponent } from './search-tips.component';

describe('SearchTipsComponent', () => {
  let component: SearchTipsComponent;
  let fixture: ComponentFixture<SearchTipsComponent>;

  beforeEach(() => {
    TestBed.configureTestingModule({
      declarations: [SearchTipsComponent],
    }).compileComponents();

    fixture = TestBed.createComponent(SearchTipsComponent);
    component = fixture.componentInstance;
  });

  it('should render tips', () => {
    component.searchTips = ['tip1', 'tip2'];
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });
});
