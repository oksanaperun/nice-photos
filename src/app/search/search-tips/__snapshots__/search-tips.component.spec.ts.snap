// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchTipsComponent should render tips 1`] = `
<app-search-tips
  searchTips={[Function Array]}
>
  <div
    class="search-tips"
  >
    
    
    <h4>
      Search tips:
    </h4>
    
    
    <ul>
      
      
      <li>
        Check your spelling and try again
      </li>
      
      
      <li>
        Try a similar but different search term
      </li>
      
      
      <li>
        Be less specific in your wording for a wider search result
      </li>
      
    
    </ul>
    
  
  </div>
  
</app-search-tips>
`;
